hphp/runtime/vm/jit/smashable-instr-arm.cpp

```diff
@@ -198,7 +198,20 @@ void smashJcc(TCA inst, TCA target, ConditionCode cc) {
 ///////////////////////////////////////////////////////////////////////////////
 
 uint64_t smashableMovqImm(TCA inst) {
-  return *reinterpret_cast<uint64_t*>(inst + smashableMovqLen() - 8);
+  using namespace vixl;
+
+  Instruction* ldr = Instruction::Cast(inst);
+  Instruction* b = ldr->NextInstruction();
+  Instruction* target = b->NextInstruction();
+  Instruction* after = target->NextInstruction()->NextInstruction();
```

Comment:
Everything that's only used in the assertx needs to be declared `DEBUG_ONLY`, e.g.

```
DEBUG_ONLY auto const ldr = Instruction::Cast(inst);
...
```